api-docs: Update `subscription` event for `op: peer_add` in feature level 205.

laurynmm opened the pull request:
In commit ada2991f1c, when a user gains access to a stream due to a role change, in addition to sending `stream op: create` events, `subscription op: peer_add` events are sent for streams that the user gains access to due to their role change.

Updates the API changelog entry for feature level 205 and the `subcription op: peer_add` event documentation for this change.

**Notes**:
- I updated the `user_ids` and `stream_ids` field descriptions in the event documentation for the role change case as the general case description wasn't quite accurate.
- This API change has been merged, but I don't know if it warrants a discussion in **#api design** since the definition of the `subcription op: peer_add` event has changed a bit.

@sahil839 mentioning you for a review of these documentation udpates, and for your thoughts on whether to start a discussion on CZO.

---

**Screenshots and screen captures:**

<details>
<summary>API changelog - feature level 205</summary>

[Current documentation](https://chat.zulip.org/api/changelog)
![Screenshot from 2023-09-05 13-39-11](https://github.com/zulip/zulip/assets/63245456/562b41c3-c208-4b9c-adc3-63c2555c162a)
</details>
<details>
<summary>Events: subscription op: peer_add</summary>

[Current documentation](https://zulip.com/api/get-events#subscription-peer_add)
![Screenshot from 2023-09-05 13-39-26](https://github.com/zulip/zulip/assets/63245456/8a19d97a-739f-4a63-acfe-0550c0001242)
</details>

---

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

sahil839 commented:
The changes look good.

Regarding using `peer_add` event here, I think it is the similar case where we use `stream op:create` event, where stream is not actually created but users comes to know that the stream is present. But we can still discuss this, and I am actually working on sending these events when stream privacy changes, so might be good to discuss it and confirm whether we need to do in a different way.

sahil839 commented:
I will start.

timabbott commented:
Discussed in https://chat.zulip.org/#narrow/stream/378-api-design/topic/peer_add.20event.20for.20users.20gaining.20access.20to.20stream/near/1635662.

laurynmm commented:
**Updates**:
- Updated for points discussed in CZO [here](https://chat.zulip.org/#narrow/stream/378-api-design/topic/peer_add.20event.20for.20users.20gaining.20access.20to.20stream/near/1636144). Specifically ...
  - Generalized `subscription` event description for `op: "peer_add"`.
  - Added a bullet point to API changelog feature level 134, for fixing the server bug of not sending `stream` events for `op: "create"` and `subscription` events for `op: "peer_add"` when a private stream was made public.

**Updated screenshots**:

<details>
<summary>API changelog updates - feature levels 205 and 134</summary>

[Current documentation](https://zulip.com/api/changelog)
![Screenshot from 2023-09-12 13-36-37](https://github.com/zulip/zulip/assets/63245456/fddef9ee-c74b-4af7-8c3a-441b372bf091)
![Screenshot from 2023-09-12 13-31-58](https://github.com/zulip/zulip/assets/63245456/2c6148ff-274e-4f79-a521-4a390aa62bc5)
</details>
<details>
<summary>subscription op: peer_add</summary>

[Current documentation](https://zulip.com/api/get-events#subscription-peer_add)
![Screenshot from 2023-09-12 13-08-28](https://github.com/zulip/zulip/assets/63245456/e48ac7e4-7303-4860-95df-766977ac7d76)
</details>
<details>
<summary>stream op: create - main description</summary>

[Current documentation](https://zulip.com/api/get-events#stream-create)
![Screenshot from 2023-09-12 13-12-11](https://github.com/zulip/zulip/assets/63245456/9bb69f1e-701e-4c8e-8499-2c09d4a6950a)
</details>

gnprice commented:
LGTM! Thanks for catching these and filling them in.

laurynmm commented:
@gnprice - Thanks for the feedback and discussion on CZO! Going to mark for integration review and remove api-docs review label.

timabbott commented:
Merged, thanks @laurynmm and @gnprice!
